execute/executetest/transformation.go

```diff
@@ -261,29 +261,59 @@ func ProcessBenchmarkHelper(
 ) {
 	b.Helper()
 
+	alloc := &memory.Allocator{}
+	input, err := genInput(alloc)
+	if err != nil {
+		b.Fatalf("error generating input: %s", err)
+	}
+
+	var tables []flux.BufferedTable
+	if err := input.Do(func(table flux.Table) error {
+		buf, err := execute.CopyTable(table)
+		if err != nil {
+			return err
+		}
+		tables = append(tables, buf)
+		return nil
+	}); err != nil {
+		b.Fatalf("error processing input tables: %s", err)
+	}
+
+	b.ResetTimer()
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		RunBenchmark(b, tables, create, alloc)
+	}
+}
+
+func RunBenchmark(
+	b *testing.B,
+	tables []flux.BufferedTable,
+	create func(id execute.DatasetID, alloc *memory.Allocator) (execute.Transformation, execute.Dataset),
+	alloc *memory.Allocator,
+) {
+	b.Helper()
+
 	defer func() {
 		if err := recover(); err != nil {
 			debug.PrintStack()
 			b.Fatalf("caught panic: %v", err)
 		}
 	}()
 
-	alloc := &memory.Allocator{}
 	parentID := RandomDatasetID()
-	tables, err := genInput(alloc)
-	if err != nil {
-		b.Fatalf("unexpected error: %s", err)
-	}
-
 	store := NewDevNullStore()
 	tx, d := create(RandomDatasetID(), alloc)
 	d.SetTriggerSpec(plan.DefaultTriggerSpec)
 	d.AddTransformation(store)
 
-	if err := tables.Do(func(table flux.Table) error {
```

Comment:
It's because there has to be a final destination. Usually it would be a result for a yield, but the dev null store acts as an end to the graph that doesn't do anything.